### PR TITLE
#36 Use bufstream crate

### PR DESCRIPTION
**Pull request by maximg**
It seems to be a good idea to propagate all changes (maybe except for very minor ones) through pull requests, so will appreciate feedback on this one. It is part of #36.

Did a quick performance check of the client and it seems to be 0.125 sec for Rust vs 0.13 for C++, similar to buf_stream.

### CODE REVIEW

File: lib/rs/Cargo.toml [hunk 1]
```diff
@@ -6,3 +6,4 @@ authors = ["Simon Génier <s@simon.coffee>", "Maxim Golov <maxim.golov@gmail.com
 
 [dependencies]
 podio = "0.1"
+bufstream = "0.1.1"
```
Comment (gsingh93) [inline lib/rs/Cargo.toml]: Might as well opt in to future non-breaking updates by making this 0.1.

File: tutorial/rs/Cargo.toml [hunk 2]
```diff
@@ -13,5 +13,8 @@ path = "src/client.rs"
 name = "server"
 path = "src/server.rs"
 
+[dependencies]
+bufstream = "0.1.1"
```
Comment (gsingh93) [inline tutorial/rs/Cargo.toml]: Same as above.

**Comment by gsingh93** [comment]
:+1: , looks good to me other than the dependency version comment. I'm glad the API is the exact same as the in-tree bufstream. What benchmark code are you using?

**Comment by in-the-mood-for-mov** [comment]
Is this the only blocker before compatibility with rust stable?

**Comment by gsingh93** [comment]
Yup. No `#![feature(..)]` means it will build on stable as far as I know.

**Comment by in-the-mood-for-mov** [comment]
> Yup. No #![feature(..)] means it will build on stable as far as I know.

That is good news. Things are so much easier since 1.0!

**Comment by maximg** [comment]
I am using a very simple benchmark (couple of thousand pings), will push it this evening on a separate branch.

**Pull request merged**